api-lib/circleAdmin.ts

```diff
@@ -23,7 +23,12 @@ const middleware =
         session_variables: sessionVariables,
       } = composeHasuraActionRequestBody(circleIdInput).parse(req.body);
 
-      if (sessionVariables.hasuraRole !== 'admin') {
+      if (sessionVariables.hasuraRole === 'admin') {
+        await handler(req, res);
```

Comment:
if we're invoking the handler early, we need to return from it to avoid executing any of the further code